### PR TITLE
fix bug about hostname by import fqdn

### PR DESCRIPTION
**Pull request by willson-chen**
The fdqn module is used to replace the regular implementation of PR #718

### CODE REVIEW

File: docs/requirements.txt [hunk 1]
```diff
@@ -31,6 +31,7 @@ sphinxcontrib-qthelp==1.0.3  # via sphinx
 sphinxcontrib-serializinghtml==1.1.4  # via sphinx
 sphinxcontrib-spelling==5.0.0  # via -r requirements.in
 urllib3==1.25.9           # via requests
+fqdn==1.4.0               # via requests
```
Comment (Julian) [inline docs/requirements.txt:34]: Did you add this here, or was it added by running `pip-compile`?

File: jsonschema/_format.py [hunk 2]
```diff
@@ -4,6 +4,7 @@
 import struct
 
 from jsonschema.exceptions import FormatError
+from fqdn import FQDN
```
Comment (Julian) [inline jsonschema/_format.py:7]:
```suggestion
import struct

from fqdn import FQDN

from jsonschema.exceptions import FormatError
```

File: setup.cfg [hunk 3]
```diff
@@ -37,12 +37,14 @@ format =
     rfc3987
     strict-rfc3339
     webcolors
+    fqdn
```
Comment (Julian) [inline setup.cfg:40]: Can you sort these two alphabetically into the list?

File: docs/requirements.txt (hunk 1 above)
Comment (willson-chen) [reply docs/requirements.txt]: The CI is reproting an error if I'm not adding this.
The error information is as follows:
```
docs-html run-test-pre: PYTHONHASHSEED='2481213764'
docs-html run-test: commands[0] | /Users/runner/work/jsonschema/jsonschema/.tox/docs-html/bin/python -m sphinx -b html /Users/runner/work/jsonschema/jsonschema/docs/ /Users/runner/work/jsonschema/jsonschema/.tox/docs-html/tmp/build -a -n -q -T -W

Traceback (most recent call last):
  File "/Users/runner/work/jsonschema/jsonschema/.tox/docs-html/site-packages/sphinx/config.py", line 319, in eval_config_file
    execfile_(filename, namespace)
  File "/Users/runner/work/jsonschema/jsonschema/.tox/docs-html/site-packages/sphinx/util/pycompat.py", line 81, in execfile_
    exec(code, _globals)
  File "/Users/runner/work/jsonschema/jsonschema/docs/conf.py", line 10, in <module>
    import jsonschema
  File "/Users/runner/work/jsonschema/jsonschema/.tox/docs-html/site-packages/jsonschema/__init__.py", line 14, in <module>
    from jsonschema._format import (
  File "/Users/runner/work/jsonschema/jsonschema/.tox/docs-html/site-packages/jsonschema/_format.py", line 7, in <module>
    from fqdn import FQDN
ModuleNotFoundError: No module named 'fqdn'
````
The link is as follows:
[error](https://github.com/willson-chen/jsonschema/runs/956671609?check_suite_focus=true)

Comment (Julian) [reply docs/requirements.txt]: Got it yeah so the docs will now depend on fqdn, but that file is autogenerated (by running pip-compile), which will automatically overwrite the file the next time someone runs it.

If you `pip install pip-tools` and then run `pip-compile`, you should get a properly regenerated requirements.txt.

Comment (willson-chen) [reply docs/requirements.txt]: Thank you very much for your comments, I have run the `pip-compile` command to regenerate the requirements.txt.
The following content is added:
```
cached-property==1.5.1    # via fqdn
colorama==0.4.3           # via sphinx
fqdn==1.4.0               # via -r requirements.in
```

**Comment by willson-chen** [comment]
@Julian I have revised all points based on the review comments, and I hope you will take the time to check them.

**Comment by Julian** [comment]
It looks like I sent you on a bit of a wild goose chase with the requirements file, I didn't notice what you were saying there until just now. Explaining the right way forward was actually a bit tricky though. If you're curious you can look at the commits I put on your branch to fix it another way.

Otherwise, think this is ready to merge.

Appreciated!

**Pull request merged**